roles/openshift_on_openstack_scale/tasks/dns.yml

```diff
@@ -0,0 +1,64 @@
+---
+# Read the DNS update key file.
+- name: Reading the remote {{ dns_key_file }} file
+  command: "grep secret {{ dns_key_file }}"
+  register: update_secret
+  changed_when: false
+
+# Create a regular expression to get the secret key.
+- name: Setting the regular expression to find the secret
+  set_fact:
+    # Example:        secret "8kjyy/xFvzapFwmcWwLzvA==";
+    secret_regex: 'secret "(.*)";'
+
+# Get the DNS server addresses from OpenStack.
+- name: Getting the IP address for the DNS VM ns-master
+  shell: "{{ openstack }} server show ns-master.{{ clusterid}}.{{ dns_domain }} --format value -c addresses"
+  # "ns-master-scale-ci.example.com-net=192.168.99.17, 172.21.0.129"
+  register: dns_addresses
+  changed_when: false
+
+# Parse the IP address from the addresses output.
+- name: Parsing the floating ip from the DNS VM addresses
+  set_fact:
+    cluster_dns_ip: "{{ dns_addresses['stdout'].split(', ')[1] }}"
+  failed_when: cluster_dns_ip == ""
+
+# Set the DNS secret value.
+- name: Creating a variable with the DNS secret value
+  set_fact:
+    dns_key_secret: "{{ update_secret['stdout']|regex_findall(secret_regex)|first }}"
+
+- name: "Remove DNS A records"
```

Comment:
As I understand we only need to remove the records if we scale down and back up. I would like to eliminate the scale down operation for now, possibly using an OpenShift reset as the scale down option. Otherwise this remove code is called for each node on scale up operations.

Reply:
Let us say, your target is 2000 openshift nodes.  Imagine you invested ~20 hours to go to 1600 nodes and your scaleup to 1700 nodes fails, because one node was unavailable for some reason.  Do you really want to reset the environment and start all over again?